### PR TITLE
Migrate Oracle and MemSQL to BaseCaseInsensitiveMappingTest and some cleanups

### PR DESCRIPTION
**Pull request by hashhar**
## Description

Some test cleanups.

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseCaseInsensitiveMappingTest.java [hunk 1]
```diff
@@ -115,6 +116,15 @@ public void testNonLowerCaseTableName()
...
 
+    /**
+     * Must return a non-empty optional with the FROM clause to select from a dual table for databases that don't allow a SELECT without a FROM clause like Oracle.
+     * Must return an empty optional for databses which don't require a FROM clause with a SELECT.
+     */
+    protected Optional<String> optionalFromDual()
+    {
+        return Optional.empty();
+    }
+
```
[the remark this answers is not in the document]
Comment (losipiuk) [reply plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseCaseInsensitiveMappingTest.java:127]: It does the job and is simple. Good to go IMO.

**Review by losipiuk — APPROVED** [review]

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseCaseInsensitiveMappingTest.java [hunk 2]
```diff
@@ -115,6 +116,15 @@ public void testNonLowerCaseTableName()
         }
     }
 
+    /**
+     * Must return a non-empty optional with the FROM clause to select from a dual table for databases that don't allow a SELECT without a FROM clause like Oracle.
+     * Must return an empty optional for databses which don't require a FROM clause with a SELECT.
```
Comment (grantatspothero) [inline plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseCaseInsensitiveMappingTest.java:121]: nit, spelling.

**Review by grantatspothero — APPROVED** [review]
👍

**Comment by hashhar** [comment]
CI hit https://github.com/trinodb/trino/issues/11130

**Pull request merged**